lib/documentmanager.js

```diff
@@ -517,7 +517,12 @@
         
         // clear and leave if that option was set
         if (this._clearCacheOnChange) {
-            this._removeDocument(id);
+            if (!change.metaDataOnly) {
+                this._logger.debug("handling imageChanged with clearCacheOnChange, removing document from cache");
```

Comment:
I left these debug logs in here on purpose since this is a hairy part of the code and I highly doubt that we've uncovered all of the possible concurrency issues.  They will not pollute the logs unless a developer explicitly sets the logging level to verbose.